src/pg_cron.c

```diff
@@ -1267,5 +1561,536 @@ ManageCronTask(CronTask *task, TimestampTz currentTime)
+					pq_parse_errornotice(&msg, &edata);
+
+					if (CronLogRun)
+					{
+						update_result = InsertOrUpdateJobRunDetail(task->runId, NULL, NULL, NULL,
+												NULL, NULL, "failed", edata.message,
+												NULL, &end_time);
+						Assert(update_result);
+					}
+
+					ereport(LOG, (errmsg("cron job " INT64_FORMAT " NOTICE: %s",
+									 task->jobId, edata.message)));
+					break;
+				}
+			case 'T':
+					break;
+			case 'C':
+				{
+					const char  *tag = pq_getmsgstring(&msg);
+					char *nonconst_tag;
+					char *cmdTuples;
+
+					nonconst_tag = strdup(tag);
+
+					if (CronLogRun)
+					{
+						update_result = InsertOrUpdateJobRunDetail(task->runId, NULL, NULL, NULL,
+												NULL, NULL, "succeeded", nonconst_tag,
+												NULL, &end_time);
+						Assert(update_result);
+					}
+
+					if (CronLogStatement) {
+						cmdTuples = pg_cron_cmdTuples(nonconst_tag);
+						ereport(LOG, (errmsg("cron job " INT64_FORMAT " COMMAND completed: %s %s",
+											 task->jobId, nonconst_tag, cmdTuples)));
+					}
+
+					free(nonconst_tag);
+					break;
+				}
+			case 'A':
+			case 'D':
+			case 'G':
+			case 'H':
+			case 'W':
+			case 'Z':
+					break;
+			default:
+					elog(WARNING, "unknown message type: %c (%zu bytes)",
+						 msg.data[0], nbytes);
+					break;
+		}
+	} else {
+
+			ExecStatusType executionStatus = PQresultStatus(result);
+			switch (executionStatus)
+			{
+				case PGRES_COMMAND_OK:
+				{
+					char *cmdStatus = PQcmdStatus(result);
+					char *cmdTuples = PQcmdTuples(result);
+
+					if (CronLogRun)
+					{
+
+						update_result = InsertOrUpdateJobRunDetail(task->runId, NULL, NULL, NULL,
+												NULL, NULL, "succeeded", cmdStatus,
+												NULL, &end_time);
+						Assert(update_result);
+					}
+
+					if (CronLogStatement)
+					{
+						ereport(LOG, (errmsg("cron job " INT64_FORMAT " COMMAND completed: %s %s",
+											 task->jobId, cmdStatus, cmdTuples)));
+					}
+
+					break;
+				}
+
+				case PGRES_BAD_RESPONSE:
+				case PGRES_FATAL_ERROR:
+				{
+					task->errorMessage = strdup(PQresultErrorMessage(result));
+					task->freeErrorMessage = true;
+					task->pollingStatus = 0;
+					task->state = CRON_TASK_ERROR;
+
+					if (CronLogRun)
+					{
+						update_result = InsertOrUpdateJobRunDetail(task->runId, NULL, NULL, NULL,
+												NULL, NULL, "failed", task->errorMessage,
+												NULL, &end_time);
+
+						Assert(update_result);
+					}
+
+					PQclear(result);
+
+					return;
+				}
+
+				case PGRES_COPY_IN:
+				case PGRES_COPY_OUT:
+				case PGRES_COPY_BOTH:
+				{
+					/* cannot handle COPY input/output */
+					task->errorMessage = "COPY not supported";
+					task->pollingStatus = 0;
+					task->state = CRON_TASK_ERROR;
+
+					if (CronLogRun)
+					{
+						update_result = InsertOrUpdateJobRunDetail(task->runId, NULL, NULL, NULL,
+												NULL, NULL, "failed", task->errorMessage,
+												NULL, &end_time);
+
+						Assert(update_result);
```

Comment:
Maybe InsertOrUpdateJobRunDetail could just return void and throw an error if it gets called without the extension being loaded?